### PR TITLE
Created post categories router & Fixed a lot part of the project - OT124-97

### PR DESCRIPTION
**Pull request by dhsosa98**
### Summary

> Created

- Created categories post router
- Created create category controller
- Created categories yup validation schema

> Updated

- Updated user seeders to store encrypted password
- Updated error handler controller
- Updated Create testimonial controller

> Fixed

- Fixed login controller
- Fixed generateJWT helper
- Fixed isRole Middleware
- Fixed validateToken & matchCredentials middlewares
- Fixed categories model & migration

> Added

- Added user role column in users model & migration
- Added categories router
- Added defaultValue 'user' to user role column in users model & migration
- Added login controller to auth controller

> Other
- Deleted controllers/login.js
- Changed matchPassword Middleware name to matchCredentials
- Refactor categories put router
- Tested all & works fine

### CODE REVIEW

File: src/models/users.js [hunk 1]
```diff
@@ -28,6 +28,10 @@ module.exports = (sequelize, DataTypes) => {
       password: {
         type: DataTypes.STRING,
         allowNull: false,
+      }, 
+      role: {
+        type: DataTypes.STRING,
+        allowNull: false,
```
Comment (rafalynch) [inline src/models/users.js:34]: Maybe we should set default value to "user".

File: src/middlewares/isRole.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
-const isAdmin = () => async (req, res, next) => {
+const isAdmin = async (req, res, next) => {
   try {
+    console.log(req.role)
```
Comment (rafalynch) [inline src/middlewares/isRole.js]:
```suggestion
    
```

**Review by rafalynch — APPROVED** [review]
LGTM! Great work!

File: src/models/users.js (hunk 1 above)
Comment (dhsosa98) [reply src/models/users.js:34]: Fixed

File: src/middlewares/isRole.js (hunk 2 above)
Comment (dhsosa98) [reply src/middlewares/isRole.js]: Fixed

**Review by panchoarc — APPROVED** [review]
Great work, lgtm to continue the normal workflow.

**Pull request merged**